### ISSUE TITLE
go to definition: don't always jump screen

### ISSUE THREAD

**Issue by kzhui125** [zed-industries/zed#10738]
### Check for existing issues

- [X] Completed

### Describe the feature

VSCode: `go to definition` rarely jump screen

<img width="486" alt="image" src="https://github.com/zed-industries/zed/assets/11925432/a13a8129-baa0-4a29-b812-d8754d7541ae">

Zed: the view changes, hard to find previous text:

before go to definition:

<img width="502" alt="image" src="https://github.com/zed-industries/zed/assets/11925432/12a67bbe-2418-4270-a8ca-aabc43c63a09">

after go to definition:

<img width="429" alt="image" src="https://github.com/zed-industries/zed/assets/11925432/8e506f89-5210-4ac7-9bed-0fda15ffa997">

```rs
pub fn add(left: usize, right: usize) -> usize {
    left + right
}

/*











*/

#[cfg(test)]
mod tests {
    #[test]
    fn it_works() {
        let name = "1";
        println!("{name}");
    }
}
```

**Comment by Moshyfawn** [comment]
There are a number of issue that suggest similar behaviour, but aren't quite the same. If you feel that one of these matches the behaviour you're seeing, we can close it in favour of one of these existing ones.

Related to:
- #5270
- #5000
- #4900

**Comment by kzhui125** [comment]
@Moshyfawn different issues

**Comment by JosephTLyons** [comment]
@kzhui125 - is the request here that if the item you are jumping to is already within view, don't move the editor at all?

**Comment by kzhui125** [comment]
yeah, don”t always scroll editor

**Comment by osa1** [comment]
> @kzhui125 - is the request here that if the item you are jumping to is already within view, don't scroll the editor at all?

Exactly. Here's a recording that demonstrates the problem. The use site and definition are next to each other, but "go to definition" scrolls dozens of lines instead of just moving cursor one line up.

https://github.com/user-attachments/assets/c9c7a7cc-c544-4fe5-afca-5e21faa898a1

**Comment by osa1** [comment]
This issue is beyond frustrating when working with a large screen.. I attempted to fix this in #18442 but I'm not sure if the code is intentional and my change breaks anything. If devs could have a look and let me know what it breaks (if anything, the tests pass) I'd like to fix this properly.
